domain/src/Entity/Domain.php

```diff
@@ -349,6 +348,21 @@ public function preSave(EntityStorageInterface $storage_controller) {
       $default->is_default = 0;
       $default->save();
     }
+    // Ensures we have a proper domain_id.
+    if ($this->isNew()) {
+      $this->createDomainId();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createDomainId() {
+    // We cannot reliably use sequences (1, 2, 3) because those can be different
+    // across environments. Instead, we use the crc32 hash function to create a
+    // unique numeric id for each domain.
+    $id = preg_replace('/[^a-z0-9_]+/', '_', $this->getHostname());
+    $this->domain_id = (int) crc32($id);
```

Comment:
If the service is available, then yes, we should fix this instance and the one above so that we have a consistent method call.